### PR TITLE
Implement auto profile indication/selection

### PR DESCRIPTION
**Pull request by gasinvein**
This adds "Auto profile" item to tray menu for enabling automatic profile selection by Tuned (the automatic profile is almost always "balanced").

### CODE REVIEW

File: src/lib/tunedmanager/tunedmanager.cpp [hunk 1]
```diff
@@ -14,6 +14,27 @@ QStringList TunedManager::GetAvailableProfiles()
     return DBusReply.value();
 }
 
+bool TunedManager::GetProfileModeAuto()
+{
+    QDBusInterface DBusInterface(BusName, BusPath, BusInterface, DBusInstance);
+    QDBusReply<QTunedProfileMode> DBusReply = DBusInterface.call(BusProfileMode);
+    const QString mode = DBusReply.value().Mode;
```
Comment (xvitaly) [inline src/lib/tunedmanager/tunedmanager.cpp:21]: Please use switch() here.

File: src/tray/trayicon/trayicon.cpp [hunk 2]
```diff
@@ -95,16 +95,41 @@ QMenu* TrayIcon::createTrayIconMenu()
     connect(quitAction, SIGNAL(triggered()), this, SLOT(exitEvent()));
 
     // Setting system tray's icon menu...
+    autoProfile = new QAction(tr("Auto profile"), this);
+    autoProfile -> setCheckable(true);
+    autoProfile -> setChecked(tunedManager -> GetProfileModeAuto());
+    if (autoProfile->isChecked())
+        autoProfile->setDisabled(true);
```
Comment (xvitaly) [inline src/tray/trayicon/trayicon.cpp:102]: It always should be disabled IMO.

File: src/tray/trayicon/trayicon.cpp [hunk 3]
```diff
@@ -95,16 +95,41 @@ QMenu* TrayIcon::createTrayIconMenu()
...
+        else
+            autoProfile->setDisabled(true);
+    }
+}
+
 void TrayIcon::profileSelectedEvent(QAction* action)
 {
     if (!tunedManager -> SetActiveProfile(action -> data().toString()))
         trayIcon -> showMessage(tr("Profile change"), tr("Failed to send the D-Bus event!"), QSystemTrayIcon::Critical);
+    else
```
Comment (xvitaly) [inline src/tray/trayicon/trayicon.cpp:128]: Don't needed.

File: src/tray/trayicon/trayicon.cpp (hunk 2 above)
Comment (gasinvein) [reply src/tray/trayicon/trayicon.cpp:102]: If it's initially unchecked (i.e. profile was selected manually before launch), it should be checkable.

File: src/lib/tunedmanager/tunedmanager.cpp [hunk 4]
```diff
@@ -14,6 +14,27 @@ QStringList TunedManager::GetAvailableProfiles()
     return DBusReply.value();
 }
 
+bool TunedManager::GetProfileModeAuto()
```
Comment (xvitaly) [inline src/lib/tunedmanager/tunedmanager.cpp:17]: IsProfileModeAuto is better I think.

File: src/lib/tunedmanager/tunedmanager.h [hunk 5]
```diff
@@ -25,6 +27,8 @@ class TunedManager : public QObject
     const QString BusProfileNameActive = "active_profile";
     const QString BusProfileNameAvailable = "profiles";
     const QString BusProfileNameSwitch = "switch_profile";
+    const QString BusProfileMode = "profile_mode";
```
Comment (xvitaly) [inline src/lib/tunedmanager/tunedmanager.h:30]: BusProfileNameMode

File: src/lib/tunedmanager/tunedmanager.h [hunk 6]
```diff
@@ -25,6 +27,8 @@ class TunedManager : public QObject
...
+    const QString BusAutoProfile = "auto_profile";
```
Comment (xvitaly) [inline src/lib/tunedmanager/tunedmanager.h:31]: BusProfileNameAuto

File: src/tray/trayicon/trayicon.cpp (hunk 2 above)
Comment (xvitaly) [reply src/tray/trayicon/trayicon.cpp:102]: OK. ACK.

File: src/tray/trayicon/trayicon.cpp (hunk 3 above)
Comment (gasinvein) [reply src/tray/trayicon/trayicon.cpp:128]: I guess the menu item should stay checkable if we failed to query DBus.

**Review by xvitaly — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**